cpp/src/arrow/csv/options.cc

```diff
@@ -43,6 +43,7 @@ ConvertOptions ConvertOptions::Defaults() {
                          "NULL", "NaN",  "n/a",      "nan",     "null"};
   options.true_values = {"1", "True", "TRUE", "true"};
   options.false_values = {"0", "False", "FALSE", "false"};
+  options.timestamp_parsers = {};
```

Comment:
Why is this necessary? `std::vector` is initialized to an empty vector by default.